library/general/src/lib/yast2/popup.rb

```diff
@@ -0,0 +1,318 @@
+require "yast"
+
+require "erb"
+
+Yast.import "Label"
+Yast.import "UI"
+
+module Yast2
+  # Class responsible for showing popups. It have small, but consistent API.
+  # Intended as replacement for Yast::Popup module.
+  # @note as UI is not easy to test, it is recommended after modifications to this class run
+  #   examples/popup_series_tester.sh which tests common combination of options.
+  # @note for rspec tests use popup_rspec where is helper for easier mocking that still does
+  # argument verifications.
+  class Popup
+    class << self
+      include Yast::I18n
+
+      # Number of lines for richtext: :auto to switch to richtext widget
+      LINES_THRESHOLD = 20
+
+      RICHTEXT_WIDTH = 60
+      RICHTEXT_HEIGHT = 10
+
+      # Shows popup and return which symbol of pressed button
+      # @param message [String] message to show
+      # @param details [String] hidden details that can be shown, if empty then it is nil shown
+      # @param headline [String, :error, :warning] sets popup headline. When String is passed it is shown as it is.
+      #   When empty string is passed no headline is shown.
+      #   If Symbol is passed it have to be from predefined set of symbols.
+      #   Note: Symbol it is just predefined strings, does not modify style of popup.
+      # @param timeout [Integer] how long wait till autoclose dialog. 0 means wait forever.
+      # @param buttons [Hash<Symbol, String>, Symbol] specified which buttons popup will have.
+      #   it can be hash in format button_id => button_text, showed in same order as in hash.
+      #   Beware that :details and :stop id is reserved.
+      #   The second option is symbol that specify one of predefined set of buttons.
+      #   Current set of predefined buttons are
+      #   `:ok` -> `{ ok: Label.OKButton}`
+      #   `continue_cancel` ->   `{ continue: Label.ContinueButton, cancel: Label.CancelButton }`
+      #   `:yes_no` -> `{ yes: Label.YesButton, no: Label.NoButton }`
+      # @param focus [Symbol, nil] what button focus.
+      #   Also it is button which is returned if timeout exceed.
+      #   if it is nil, then choose the first button. See buttons parameter.
+      # @param richtext [Boolean, :auto] if use richtext widget.
+      #   Useful when report contain richtext tags or is long, so it have to be scrollable.
+      #   If :auto is used, then it detect too long text and use richtext for it and! escape
+      #   richtext tags, so if text is really richtext then use true.
+      # @param style [:notice, :important, :warning] popup dialog styling. :notice is common one,
```

Comment:
Huh, it still does not tell me the color :-( Can we say, "important is green, warning is red"?

Reply:
it is defined in style sheet. To be honest I do not know and I think it is different in SLE installer and in runtime.